r2i/onnxrt/parameters.h

```diff
@@ -40,89 +41,21 @@ class Parameters: public IParameters {
   RuntimeError Set (const std::string &in_parameter, int in_value) override;
   RuntimeError Set (const std::string &in_parameter, double in_value) override;
   RuntimeError List (std::vector<ParameterMeta> &metas) override;
+  /* To set/get Parameters of the internal Engine instance */
+  RuntimeError SetLogLevel (int &value);
+  RuntimeError GetLogLevel (int &value);
+  RuntimeError SetIntraNumThreads (int &value);
+  RuntimeError GetIntraNumThreads (int &value);
+  RuntimeError SetGraphOptLevel (int &value);
+  RuntimeError GetGraphOptLevel (int &value);
+  RuntimeError SetLogId (std::string &value);
+  RuntimeError GetLogId (std::string &value);
 
- private:
-  std::shared_ptr <Engine> engine;
-
-  friend class Accessor;
-
-  class Accessor {
-   public:
-    Accessor (Parameters *target) : target(target) {}
-    virtual RuntimeError Set () = 0;
-    virtual RuntimeError Get () = 0;
-    virtual ~Accessor() {}
-
-   protected:
-    Parameters *target;
-  };
-
-  class StringAccessor : public Accessor {
-   public:
-    StringAccessor (Parameters *target) : Accessor(target) {}
-    std::string value;
-  };
-
-  class IntAccessor : public Accessor {
-   public:
-    IntAccessor (Parameters *target) : Accessor(target) {}
-    int value;
-  };
-
-  class LoggingLevelAccessor : public IntAccessor {
-   public:
-    LoggingLevelAccessor (Parameters *target) : IntAccessor(target) {}
-
-    RuntimeError Set () {
-      return target->engine->SetLoggingLevel(this->value);
-    }
-
-    RuntimeError Get () {
-      this->value = target->engine->GetLoggingLevel();
-      return RuntimeError ();
-    }
-  };
-
-  class LogIdAccessor : public StringAccessor {
-   public:
-    LogIdAccessor (Parameters *target) : StringAccessor(target) {}
-    RuntimeError Set () {
-      return target->engine->SetLogId(this->value);
-    }
-
-    RuntimeError Get () {
-      this->value = target->engine->GetLogId();
-      return RuntimeError ();
-    }
-  };
-
-  class IntraNumThreadsAccessor : public IntAccessor {
-   public:
-    IntraNumThreadsAccessor (Parameters *target) : IntAccessor(target) {}
-
-    RuntimeError Set () {
-      return target->engine->SetIntraNumThreads(this->value);
-    }
-
-    RuntimeError Get () {
-      this->value = target->engine->GetIntraNumThreads();
-      return RuntimeError ();
-    }
-  };
-
-  class GraphOptLevelAccessor : public IntAccessor {
-   public:
-    GraphOptLevelAccessor (Parameters *target) : IntAccessor(target) {}
-
-    RuntimeError Set () {
-      return target->engine->SetGraphOptLevel(this->value);
-    }
+ protected:
+  std::shared_ptr <IEngine> engine;
```

Comment:
Why are you saving the interface and not the specific class?

Reply:
Solved [here](https://github.com/RidgeRun/r2inference/commit/63f9657aa4f24652bc8298afad787bf87379eaa5)